src/net_processing.cpp

```diff
@@ -1632,7 +1632,7 @@ void PeerManagerImpl::BlockDisconnected(const std::shared_ptr<const CBlock> &blo
  */
 void PeerManagerImpl::NewPoWValidBlock(const CBlockIndex *pindex, const std::shared_ptr<const CBlock>& pblock)
 {
-    std::shared_ptr<const CBlockHeaderAndShortTxIDs> pcmpctblock = std::make_shared<const CBlockHeaderAndShortTxIDs> (*pblock, true);
+    std::shared_ptr<const CBlockHeaderAndShortTxIDs> pcmpctblock = std::make_shared<const CBlockHeaderAndShortTxIDs> (*pblock);
```

Comment:
```suggestion
    auto pcmpctblock{std::make_shared<const CBlockHeaderAndShortTxIDs>(*pblock)};
```

nit: Or maybe even use `auto` :sweat_smile:

Reply:
nit: Or maybe even inline into lazy_ser to avoid constructing it when there is no need to. (Might minimally speed up IBD?)

Reply:
I've changed to `auto`.

> Or maybe even inline into lazy_ser to avoid constructing it when there is no need to. (Might minimally speed up IBD?)

I'm not sure how this would work. `pcmpctblock` is required further down the function outside the lazy_ser lambda. I have moved the instantiation below the early exits, which I think addresses your point about speeding up IBD.

Reply:
Yeah, I missed that assignment. Seems fine to just move it down for now.